### PR TITLE
ISPN-12070 s/whitelist/allowlist/

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-12070

### CODE REVIEW

File: commons/all/src/main/java/org/infinispan/commons/configuration/ClassAllowList.java [hunk 1]
```diff
@@ -24,23 +24,27 @@
...
- * and fully qualified class names from "infinispan.deserialization.whitelist.classes".
+ * By default it includes regular expressions from the system property "infinispan.deserialization.allowlist.regexps"
+ * and fully qualified class names from "infinispan.deserialization.allowlist.classes".
  * <p>
  * Classes are checked first against the set of class names, and in case not present each of the regular expressions are
  * evaluated in the order supplied.
  *
  * @since 9.4
  */
-public final class ClassWhiteList {
```
Comment (ryanemerson) [inline commons/all/src/main/java/org/infinispan/commons/configuration/ClassAllowList.java:38]: As this is a public package, I think we need to keep `ClassWhiteList` until 14. We can just make it  extend `ClassAllowList`.

File: commons/all/src/main/java/org/infinispan/commons/dataconversion/JavaSerializationEncoder.java [hunk 2]
```diff
@@ -12,8 +12,8 @@
 @Deprecated
 public class JavaSerializationEncoder extends MarshallerEncoder {
 
-   public JavaSerializationEncoder(ClassWhiteList classWhiteList) {
-      super(new JavaSerializationMarshaller(classWhiteList));
+   public JavaSerializationEncoder(ClassAllowList classAllowList) {
```
Comment (ryanemerson) [inline commons/all/src/main/java/org/infinispan/commons/dataconversion/JavaSerializationEncoder.java:15]: If we keep `ClassWhiteList` and make it extend `ClassAllowList`, this ctor should remain the same for backwards compatibility and we can just pass it to `JavaSerializationMarshaller`.

File: core/src/main/java/org/infinispan/configuration/global/AllowListConfigurationBuilder.java [hunk 3]
```diff
@@ -11,55 +11,55 @@
...
 import org.infinispan.commons.configuration.attributes.AttributeSet;
 
 /**
- * Configures the {@link org.infinispan.manager.EmbeddedCacheManager} {@link org.infinispan.commons.configuration.ClassWhiteList}.
+ * Configures the {@link org.infinispan.manager.EmbeddedCacheManager} {@link ClassAllowList}.
  *
  * @author Ryan Emerson
  * @since 10.0
  */
-public class WhiteListConfigurationBuilder implements Builder<WhiteListConfiguration> {
```
Comment (ryanemerson) [inline core/src/main/java/org/infinispan/configuration/global/AllowListConfigurationBuilder.java:23]: We probably also need to keep this for backwards compatibility in Embedded mode.

File: core/src/main/java/org/infinispan/configuration/parsing/Parser.java [hunk 4]
```diff
@@ -172,9 +172,11 @@ private void parseSerialization(final XMLExtendedStreamReader reader, final Conf
                parseSerializationContextInitializer(reader, holder.getClassLoader(), builder.serialization());
                break;
             }
-            case WHITE_LIST: {
+            case WHITE_LIST:
+               CONFIG.elementDeprecatedUseOther(Element.WHITE_LIST, Element.ALLOW_LIST);
```
Comment (ryanemerson) [inline core/src/main/java/org/infinispan/configuration/parsing/Parser.java:176]: If the 12.0 schema is used shouldn't this thrown unrecognised attribute exception?

**Comment by tristantarrant** [comment]
@ryanemerson I believe I've addressed all issues

File: commons/all/src/main/java/org/infinispan/commons/configuration/ClassWhiteList.java [hunk 5]
```diff
@@ -1,153 +1,8 @@
...
-
-   public void read(ClassWhiteList whiteList) {
-      this.regexps.addAll(whiteList.regexps);
-      this.compiled.addAll(whiteList.compiled);
-      this.classes.addAll(whiteList.classes);
-   }
+ * @deprecated since 12.0. Will be removed in 14.0. Use {@link ClassAllowList}.
+ **/
+@Deprecated
+public final class ClassWhiteList extends ClassAllowList {
```
Comment (ryanemerson) [inline commons/all/src/main/java/org/infinispan/commons/configuration/ClassWhiteList.java:7]: We still need to keep the old constructors for backwards compatibility:

```java
   public ClassWhiteList() {
      super();
   }

   public ClassWhiteList(List<String> regexps) {
      super(regexps);
   }

   public ClassWhiteList(Collection<String> classes, List<String> regexps) {
      super(classes, regexps);
   }
```

File: commons/all/src/main/java/org/infinispan/commons/configuration/ClassAllowList.java (hunk 1 above)
Comment (tristantarrant) [reply commons/all/src/main/java/org/infinispan/commons/configuration/ClassAllowList.java:38]: Done

File: core/src/main/java/org/infinispan/configuration/parsing/Parser.java (hunk 4 above)
Comment (tristantarrant) [reply core/src/main/java/org/infinispan/configuration/parsing/Parser.java]: Done

File: commons/all/src/main/java/org/infinispan/commons/configuration/ClassWhiteList.java (hunk 5 above)
Comment (tristantarrant) [reply commons/all/src/main/java/org/infinispan/commons/configuration/ClassWhiteList.java:10]: Done

**Comment by ryanemerson** [comment]
A lot of the CI failures appear to be related.

**Comment by ryanemerson** [comment]
```java
 12:06:38.803 [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:testCompile (default-testCompile) on project infinispan-core: Compilation failure
17:06:40  12:06:38.803 [ERROR] /home/infinispan/workspace/Infinispan_PR-8541/core/src/test/java/org/infinispan/container/versioning/WriteSkewConsistencyTest.java:[147,72] org.infinispan.util.ControlledConsistentHashFactory is abstract; cannot be instantiated
17:06:40  12:06:38.803 [ERROR] -> [Help 1]
17:06:40  org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:testCompile (default-testCompile) on project infinispan-core: Compilation failure
17:06:40  /home/infinispan/workspace/Infinispan_PR-8541/core/src/test/java/org/infinispan/container/versioning/WriteSkewConsistencyTest.java:[147,72] org.infinispan.util.ControlledConsistentHashFactory is abstract; cannot be instantiated
17:06:40  
17:06:40      at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
17:06:40      at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
17:06:40      at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
17:06:40      at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
17:06:40      at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
17:06:40      at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
17:06:40      at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
17:06:40      at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
17:06:40      at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
17:06:40      at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
17:06:40      at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
17:06:40      at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
17:06:40      at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
17:06:40      at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
17:06:40      at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
17:06:40      at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
17:06:40      at java.lang.reflect.Method.invoke (Method.java:566)
17:06:40      at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
17:06:40      at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
17:06:40      at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
17:06:40      at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
17:06:40  Caused by: org.apache.maven.plugin.compiler.CompilationFailureException: Compilation failure
17:06:40  /home/infinispan/workspace/Infinispan_PR-8541/core/src/test/java/org/infinispan/container/versioning/WriteSkewConsistencyTest.java:[147,72] org.infinispan.util.ControlledConsistentHashFactory is abstract; cannot be instantiated
```

**Comment by ryanemerson** [comment]
Thanks @tristantarrant

**Pull request merged**